### PR TITLE
Cooking plugin

### PR DESCRIPTION
**Pull request by Cronick**
As talked about on discord. I have tried to create a basic cooking plugin.
Basic overlay, and there is also burn procentage for "Burnt" food.

image

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/cooking/CookingOverlay.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import net.runelite.client.ui.overlay.components.LineComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.components.TitleComponent;
+
+import static net.runelite.api.AnimationID.COOKING_FIRE;
+import static net.runelite.api.AnimationID.COOKING_RANGE;
+
+class CookingOverlay extends Overlay
+{
+    private static final int[] animationIds =
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/cooking/CookingOverlay.java]: This should be a hashset

File: runelite-client/src/main/java/net/runelite/client/plugins/cooking/CookingOverlay.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		{
+			panelComponent.getChildren().add(TitleComponent.builder()
+				.text("NOT cooking")
+				.color(Color.RED)
+				.build());
+		}
+
+		panelComponent.getChildren().add(LineComponent.builder()
+			.left("Cooked:")
+			.right(session.getCookAmount() + (session.getCookAmount() >= 1 ? " (" + xpTrackerService.getActionsHr(Skill.COOKING) + "/hr)" : ""))
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/cooking/CookingOverlay.java]: This would throw whenever you aren't cooking.

Comment (Cronick) [reply runelite-client/src/main/java/net/runelite/client/plugins/cooking/CookingOverlay.java]: Yeah. I forgot about that part, and I can see that you have provided a better solution for a good amount of the code, so should have been fixed 👍

**Comment by Adam-** [comment]
This is still broken when cooking on fires since the animation isn't long enough it alternates between cooking and not cooking.

**Comment by Cronick** [comment]
> This is still broken when cooking on fires since the animation isn't long enough it alternates between cooking and not cooking.

Added to the if statement for isCooking() a delay checking, so if there is over 3 seconds between each animation it will say "Not cooking"

Took a look at client.getLocalPlayer().getAnimation(), and it keeps saying going to IDLE (-1) between the animation for some weird reason randomly, and it varies in length. So I know it's not the most beautiful solution on the issue.

**Pull request merged**